packages/dx-react-grid/docs/guides/custom-plugin-development.md

```diff
@@ -0,0 +1,709 @@
+           }, {
+            changeToolbarFilterValue // Accessing actions
+          }) => (
+            <Select
+              ...
+              onChange={(event) => {changeToolbarFilterValue(event.target.value)}}
+            >
+              ...
+            </Select>
+          )}
+          </TemplateConnector>
+        </Template>
+      </Plugin>
+    );
+  }
+}
+```
+
+As you can see, the event handler accepts the `event` parameter that we can use to access the filter value. The filter value is passed to the `changeToolbarFilterValue` action (the previous code) and then to the state object of the `ToolbarFilterState` plugin (the code below).
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+export class ToolbarFilterState extends React.PureComponent {
+  ...
+  changeValue(value) {
+    this.setState({
+      filterValue: value
+    });
+  }
+  ...
+}
+```
+
+Although we are detecting changes in the select box, the grid does not apply a filter yet. We need to add the [IntegratedFiltering](../reference/integrated-filtering.md) plugin to the Grid. According to the docs, the plugin imports a `filterExpression` getter that represents the filter expression to be applied. We should use a function to extend the `filterExpression` with the selected value.
+
+*index.js:*
+```jsx
+...
+import {
+  IntegratedFiltering
+} from '@devexpress/dx-react-grid';
+
+class App extends Component {
+  ...
+  render() {
+    ...
+    return (
+      <Paper>
+        <Grid
+          ...
+        >
+          <ToolbarFilterState columnName="CompanyName" defaultFilterValue=""/>
+          <IntegratedFiltering />
+          <Table />
+          <TableHeaderRow />
+          <Toolbar />
+          <ToolbarFilter />
+        </Grid>
+      </Paper>
+    );
+  }
+}
+```
+
+*file toolbar-filter-state.js:*
+```jsx
+...
+
+export class ToolbarFilterState extends React.PureComponent {
+  constructor(props) {
+    ...
+    this.filterExpressionComputed = this.filterExpressionComputed.bind(this);
+  }
+  ...
+  filterExpressionComputed({filterExpression, toolbarFilterValue}) {
+    const newFilterExpression = { ...filterExpression || {filters: [], operator: 'and'} };
+    if(toolbarFilterValue) {
+      newFilterExpression.filters = [...newFilterExpression.filters, {
+        columnName: this.props.columnName,
+        value: toolbarFilterValue
+      }];
+    }
+    return newFilterExpression;
+  }
+  render() {
+    ...
+    return (
+      <Plugin
+        name="ToolbarFilterState"
+      >
+        ...
+        <Getter name="filterExpression" computed={this.filterExpressionComputed} />
+      </Plugin>
+    );
+  }
+}
+```
+
+Now, you can filter grid rows by the `CompanyName` column.
+
+## Final Steps
+
+At the final steps, we should display the current column's caption near the select box and add the `Clear` button that clears the selected value. Since these steps do not add anything to the concepts explained earlier, we will go over them briefly:
```

Comment:
First, do you really need a section called 'Final Steps'?

Reply:
Why do you add "Since these steps do not add anything to the concepts explained earlier, we will go over them briefly"? Why future tense?